pkg/k8smgmt/ingress-nginx.go

```diff
@@ -0,0 +1,158 @@
+	IngressNginxExternalIPRetries = 30
+	IngressDefaultCertSecret      = "default-cert"
+)
+
+type RefreshCertsOpts struct {
+	CommerialCerts bool
+	InitCluster    bool // force create of secret to ensure it exists
+}
+
+// RefreshCert refreshes the specified wildcard cert for the
+// cloudlet in the cluster pointed to by the KConfNames.
+// The certificate is used as the default certificate for
+// all ingress instances in the cluster.
+func RefreshCert(ctx context.Context, client ssh.Client, names *KconfNames, cloudletKey *edgeproto.CloudletKey, cache *certscache.ProxyCertsCache, wildcardName string, opts RefreshCertsOpts) error {
+	log.SpanLog(ctx, log.DebugLevelInfra, "k8s refresh ingress certs", "cloudlet", cloudletKey, "certName", wildcardName)
+	if os.Getenv("E2ETEST_TLS") != "" {
+		opts.CommerialCerts = false
+	}
+
+	cert, updated, err := cache.RefreshCert(ctx, cloudletKey, wildcardName, opts.CommerialCerts)
+	if err != nil {
+		log.SpanLog(ctx, log.DebugLevelInfra, "k8s failed to refresh ingress certs", "certName", wildcardName, "err", err)
+		return err
+	}
+	if !updated && !opts.InitCluster {
+		return nil
+	}
+	fileName := strings.Replace(wildcardName, "*", "_", 1)
+
+	err = pc.WriteFile(client, fileName+".crt", cert.CertString, "cert", pc.NoSudo)
+	if err != nil {
+		return fmt.Errorf("failed to write cert file %s.crt, %s", fileName, err)
+	}
+	err = pc.WriteFile(client, fileName+".key", cert.KeyString, "cert", pc.NoSudo)
+	if err != nil {
+		return fmt.Errorf("failed to write cert key file %s.key, %s", fileName, err)
+	}
+	// this command generates a yaml file then applies it, to allow
+	// us to update the secret if it already exists
+	cmd := fmt.Sprintf("kubectl %s create secret -n %s tls %s --key %s.key --cert %s.crt --save-config --dry-run=client -o yaml | kubectl %s apply -f -", names.KconfArg, IngressNginxNamespace, IngressDefaultCertSecret, fileName, fileName, names.KconfArg)
+	out, err := client.Output(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to write cert secret: %s, %s", out, err)
+	}
+	return nil
+}
+
+// InstallIngressNginx installs the ingress-nginx controller
+// in the cluster.
+func InstallIngressNginx(ctx context.Context, client ssh.Client, names *KconfNames, waitForExternalIP bool) error {
+	log.SpanLog(ctx, log.DebugLevelInfra, "install ingress nginx")
+
+	// how to determine available versions:
+	// helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+	// helm search repo ingress-nginx/ingress-nginx --versions
+
+	// This specifies a default certificate, which should be a
+	// wildcard cert for the entire cluster/cloudlet.
+	cmd := fmt.Sprintf("helm %s upgrade --install %s %s --repo %s --namespace %s --create-namespace --version %s --set controller.extraArgs.default-ssl-certificate=%s/%s", names.KconfArg, IngressNginxName, IngressNginxChart, IngressNginxRepoURL, IngressNginxNamespace, IngressNginxChartVersion, IngressNginxNamespace, IngressDefaultCertSecret)
```

Comment:
why `helm upgrade` instead of `helm install`?
Also now you can use `--wait` option to make sure all pods are up, but I guess you'll still need to wait for lb ip, so that probably doesn't matter.